Users/toml_config.rst

```diff
@@ -0,0 +1,286 @@
+See the following example::
+
+  [all]
+  files = '**'
+  aspects = ['aspectname1', 'AspectName2'] # case-insensitive
+  # defining an aspect's taste
+  aspectname1.aspect_taste = 80
+  # we can define subaspect taste through its parent
+  aspectname1.subaspect_taste = ['word1', 'word2', 'word3']
+
+  ['all.python']
+  files = '**.py'
+  language = 'Python'
+  # appending additional aspect
+  appends = 'all'
+  aspects = 'aspectname3'
+  # excluding certain subaspect
+  excludes = 'AspectName2Subaspect'
+
+
+For existing coala users
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section we will see how to convert a complex coafile into
+a configuration file in TOML
+
+coafile ::
+
+ [all]
+ files = *.py, coantlib/**/*.py, tests/**/*.py, coantbears/**/*.py, .ci/*.py
+ max_line_length = 80
+ use_spaces = True
+
+ [all.python]
+ # Patches may conflict with autopep8 so putting them in own section so they
+ # will be executed sequentially; also we need the LineLengthBear to double
+ # check the line length because PEP8Bear sometimes isn't able to correct the
+ # linelength.
+ bears = SpaceConsistencyBear
+ language = Python
+ preferred_quotation = '
+
+ default_actions = **: ApplyPatchAction
+
+ [all.flakes]
+ # Do not set default_action to ApplyPatchAction as it may lead to some
+ # required imports being removed that might result in coala behaving weirdly.
+
+ default_actions = *: ShowPatchAction
+
+ bears += PyUnusedCodeBear
+ language = Python
+ remove_all_unused_imports = true
+
+To convert a coafile to configuration file in TOML
+
+- Enclose all string values in quotes
+- Use array notation to depict list of strings
+- Replace ``[parent_section.inherited_section]]`` with ``[inherited.section]``
+  and add ``inherits = parent_section`` as a key-value pair
+- Use ``true`` or ``false`` to specify booleans
+- Replace ``a += b`` with
+  ::
+
+   a = 'b'
+   appends = 'a'
+
+- If you are using aspects  ``a:b = 'c'``  in a section named `example`
+  then replace
+  ``a:b = 'c'`` with ``a.b = 'c'`` or
+  ::
+
+   [example.a]
+   b = 'c'
+
+Using the above rules we get a configuration file in TOML
+
+::
+
+ [all]
+ files = ['*.py', 'coantlib/**/*.py', 'tests/**/*.py', 'coantbears/**/*.py',
+         '.ci/*.py']
+ max_line_length = 80
+ use_spaces = true
+
+ [python]
+ # Patches may conflict with autopep8 so putting them in own section so they
+ # will be executed sequentially; also we need the LineLengthBear to double
+ # check the line length because PEP8Bear sometimes isn't able to correct the
+ # linelength.
+ inherits = 'all'
+ bears = 'SpaceConsistencyBear'
+ language = 'Python'
+ preferred_quotation = '
+
+ default_actions = '**: ApplyPatchAction'
+
+ [flakes]
+ # Do not set default_action to ApplyPatchAction as it may lead to some
+ # required imports being removed that might result in coala behaving weirdly.
+ inherits = 'all'
+ default_actions = '*: ShowPatchAction'
+
+ bears = 'PyUnusedCodeBear'
```

Comment:
Mention above that if `property` is redefined in the inherited section then it will overwrite if `appends` is not mentioned